tests/console/zfs.pm

```diff
@@ -59,11 +59,16 @@ sub check_available {
 
 sub install_zfs {
     my $repo = get_repository();
-    die "Sorry, zfs repository is not (yet) available for this distribution" unless (check_available($repo));
+    if (!check_available($repo)) {
+        my $msg = "Sorry, zfs repository is not (yet) available for this distribution";
+        record_soft_failure($msg);
```

Comment:
I really like to follow the rule "no record softfailure without bug id"  - after this PR merged the module will stay in softfailed state for undefined period of time the same as job where it is running . Which will lead to two bad things : 
1. Reviewers will get used to see this module softfailed **and** all the jobs where it is running 
2. Reviewers might miss soft-failures in **other** modules running with this one because they get used to the fact  "ah this job is yellow for ages"
So whenever it is possible to avoid soft-failure I would vote for this .  In this certain case I see such possibility in leaving this file unchanged and simply exclude zfs test from running here https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/main_common.pm#L1817

Reply:
`zfs` is a exceptional case. It is not supported from the main repositories, and despite we want to have maximum possible coverage, it should not block the Leap release process, as this is an external repository.
This is also the reason, why there is no bug id for "this repository is not yet available on Leap 15.3". This is one of the rare exceptions.

It would also not be quiet correct to exclude 15.3 from the test runs, as the repository does not exists now, but will become available in the future. The proposed change guarantees maximum coverage while it does not mark a whole test run as failed, just because a third-party module is not yet available.

I think that is a fair compromise between maximum coverage, being non-disruptive and no-effort to include this test run, when the repository becomes available.

Reply:
so basically it is about to measure what would brings more damage : 
 1. having `extra_tests_filesystem` for Leap 15.3 soft-failed for undefined period of time
 2. potentially miss moment when it will be enabled in Leap 15.3 and have such gap in coverage 
 
For me personally I would call 1 bigger just because you will get it right now while second one is just potential and there few measures could be taken to not have it at all ( e.g. you can add dummy test to `extra_tests_filesystem` which will start failing when repo become available for 15.3 ) 
But it is just my IMHO , feel free to merge it ... if you will find someone else who will give you +1 here :D